packages/adminrouter/extra/src/lib/util.lua

```diff
@@ -243,5 +243,34 @@ function string:strip()
     return self:match "^%s*(.-)%s*$"
 end
 
+local error_response_bodies = {}
+local errorpages_dir_path = os.getenv("AUTH_ERROR_PAGE_DIR_PATH")
+if errorpages_dir_path == nil then
+    ngx.log(ngx.WARN, "AUTH_ERROR_PAGE_DIR_PATH not set.")
+else
+    error_codes = {401, 403, 404, 500, 501, 502, 503, 504}
+    for _, code in ipairs(error_codes) do
+        local p = errorpages_dir_path .. "/" .. code .. ".html"
```

Comment:
Did you describe / discuss somewhere why you're not using the nginx [`error_page`](http://nginx.org/r/error_page) directive? Can a strategy as described in https://stackoverflow.com/a/13624139/145400 be used to make this cleaner?

Reply:
> Did you describe / discuss somewhere why you're not using the nginx error_page directive? 

There are three reasons:
a) consistency: 401 and 403 in some of the cases are emitted using Lua[1][2]. I was aiming at reusing the code we already have, instead of having two ways to do the same thing in AR
b) extendability: as described in [3], at some point it would be nice to unify error pages handling for other parts of the code. `/service` endpoint and others alike require a custom error string to be present in the error page which will be explaining why the error occurred in the first place. Static error pages will not cut here.
c) speed: although I can't tell for sure without solid data/benchmarks, I do not think there is going to be a significant difference when it comes to performance between static pages and the pages served from memory by JITed Lua code.

[1] https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/lib/auth/open.lua#L47
[2] https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/lib/auth/open.lua#L25
[3] https://jira.mesosphere.com/browse/DCOS_OSS-1937

> Can a strategy as described in https://stackoverflow.com/a/13624139/145400 be used to make this cleaner?

If we disregard paragraphs a) and b), then yes - it is possible.